src/components/Review/DecisionArea.tsx

```diff
@@ -1,137 +1,169 @@
-import React, { useEffect, useState } from 'react'
-import { Button, Container, Form, Header, Modal, Radio, Segment, TextArea } from 'semantic-ui-react'
+import React, { useState } from 'react'
+import { Button, Form, Header, Modal, Radio, Segment, TextArea } from 'semantic-ui-react'
 import { useRouter } from '../../utils/hooks/useRouter'
 import { ReviewResponse, ReviewResponseDecision } from '../../utils/generated/graphql'
 import strings from '../../utils/constants'
 import messages from '../../utils/messages'
 import SummaryViewWrapper from '../../formElementPlugins/SummaryViewWrapper'
 import { SummaryViewWrapperProps } from '../../formElementPlugins/types'
 import useUpdateReviewResponse from '../../utils/hooks/useUpdateReviewResponse'
+import { getOperationAST } from 'graphql'
 
-interface DecisionAreaProps {
+type DecisionAreaProps = SummaryViewWrapperProps & {
   reviewResponse: ReviewResponse
-  toggle: boolean
-  summaryViewProps: SummaryViewWrapperProps
+  isConsolidation: boolean
 }
 
 const DecisionArea: React.FC<DecisionAreaProps> = ({
-  toggle,
   reviewResponse,
-  summaryViewProps,
+  isConsolidation,
+  ...summaryViewProps
 }) => {
-  const { updateQuery } = useRouter()
-  const [isOpen, setIsOpen] = useState(false)
-  const [previousToggle, setPreviousToggle] = useState(false)
-  const [review, setReview] = useState(reviewResponse)
-  const updateResponse = useUpdateReviewResponse(reviewResponse.id)
-
-  useEffect(() => {
-    if (toggle != previousToggle) {
-      setReview(reviewResponse)
-      setPreviousToggle(toggle)
-      setIsOpen(true)
-    }
-  }, [toggle])
+  const {
+    query: { openResponse },
+    updateQuery,
+  } = useRouter()
+  const [reviewUpdate, setReviewUpdate] = useState(reviewResponse)
+  const updateResponse = useUpdateReviewResponse()
 
   const submit = async () => {
-    // TODO do we need to handle update error ?
-    await updateResponse(review)
     updateQuery({ openResponse: null })
+    // TODO do we need to handle update error ?
+    await updateResponse(reviewUpdate)
+  }
+
+  const validateDecisionComment = () =>
+    !reviewUpdate.comment && isConsolidation
+      ? reviewUpdate?.decision === ReviewResponseDecision.Disagree
+      : reviewUpdate?.decision === ReviewResponseDecision.Decline
+
+  // TODO: Improve - how to use shorthand was on my mind...
+  const getOptions = () => {
+    return isConsolidation ? (
+      <>
+        <strong>{strings.LABEL_CONSOLIDATE}</strong>
+        <Form.Field>
+          <Radio
+            label={strings.LABEL_CONSOLIDATION_AGREEMENT}
```

Comment:
could have an object describing decision and label:
```
const optionsMap = {
  consolidation: [{
    label: strings.LABEL_CONSOLIDATION_AGREEMENT,
    decision: ReviewResponseDecision.Agree
  },...],
  review:...
}
```
then 
```
const options = isConsolidation ? optionsMap.consolidation : optionsMap.review
...
   {options.map(({decision. label}) => (
   <Form.Field key={decision}>
          <Radio
            label={label}
            value={decision
            name="decisionGroup"
            checked={reviewUpdate.decision === decision}
            onChange={() =>
              setReviewUpdate({
                ...reviewUpdate,
                decision: decision,
              })
            }
          />
   )}
```

Reply:
Yes, that's a very good solution : ) I'll add that in, thanks

Reply:
Done: 7519f83